datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala

```diff
@@ -83,8 +83,16 @@ class MVAnalyzerRule(sparkSession: SparkSession) extends Rule[LogicalPlan] {
         }
         Aggregate(grp, aExp, child)
     }
-    val catalog = DataMapStoreManager.getInstance().getDataMapCatalog(dataMapProvider,
+    var catalog = DataMapStoreManager.getInstance().getDataMapCatalog(dataMapProvider,
       DataMapClassProvider.MV.getShortName).asInstanceOf[SummaryDatasetCatalog]
+    // when first time DataMapCatalogs are initialized, it stores session info also, but when carbon
```

Comment:
Can this DataMapStoreManager be initialized from CarbonEnv?? This way we can make sure that each sparksession has a new DataMapStoreManger.

Reply:
This is map stored in memory right, even if you initialize in carbonenv, every new session we need to clear and reload again. may be we might have better integration, but if we change integration test will take, may be we can take up this later

Reply:
ok